src/UserRepository/HtpasswdFactory.php

```diff
@@ -24,9 +25,10 @@ class HtpasswdFactory
      */
     public function __invoke(ContainerInterface $container): Htpasswd
     {
-        $config = $container->get('config');
-        Assert::isMap($config);
+        /** @var ArrayAccess<array-key,mixed> $config */
+        $config     = $container->get('config');
         $authConfig = $config['authentication'] ?? [];
+        Assert::isMap($authConfig);
 
         $htpasswd = $authConfig['htpasswd'] ?? null;
         Assert::nullOrString($htpasswd);
```

Comment:
```suggestion
        $config   = $container->get('config');
        Assert::isArrayAccessible($config);
        $htpasswd = $config['authentication']['htpasswd'] ?? null;
        Assert::nullOrString($htpasswd);
```
Will this work?

Reply:
If `config` may be set as `null` it would crash.
We can keep this in mind for the next major of this and probably want to add some kind of compatibility-layer in `mezzio` itself to wrap any container within a container delegator to ensure that `config` is always an array.

Gonna add this as an RFC in the mezzio project after @weierophinney merged the issue template for RFCs.